bufferserver/src/main/java/com/datatorrent/bufferserver/server/Server.java

```diff
@@ -130,27 +132,100 @@ public void run()
         storageHelperExecutor.shutdown();
         try {
           serverHelperExecutor.awaitTermination(5000, TimeUnit.MILLISECONDS);
+          storageHelperExecutor.awaitTermination(5000, TimeUnit.MILLISECONDS);
         } catch (InterruptedException ex) {
           logger.debug("Executor Termination", ex);
         }
-        logger.info("Server stopped listening at {}", address);
+        logger.info("Server stopped listening at {}", getServerAddress());
+        latch.countDown();
+        latch = null;
       }
     });
   }
 
-  public synchronized InetSocketAddress run(EventLoop eventloop)
+  public InetSocketAddress run()
   {
+    final CountDownLatch latch = new CountDownLatch(1);
+    this.latch = latch;
```

Comment:
No, it is the same pattern as final local variable passed to an anonymous class. The latch is not reusable.